fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/mixin/entity/event/effect/LivingEntityMixin.java

```diff
@@ -0,0 +1,219 @@
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
+import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
+import com.llamalad7.mixinextras.sugar.Local;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.core.Holder;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.level.Level;
+
+import net.fabricmc.fabric.api.entity.event.v1.effect.ServerMobEffectEvents;
+import net.fabricmc.fabric.impl.entity.event.effect.MobEffectUtil;
+
+@Mixin(LivingEntity.class)
+public abstract class LivingEntityMixin extends Entity {
+	private LivingEntityMixin(EntityType<?> entityType, Level level) {
+		super(entityType, level);
+	}
+
+	@WrapMethod(method = "canBeAffected")
+	private boolean allowAddEffect(MobEffectInstance effectInstance, Operation<Boolean> original) {
+		if (this.fabric_isClient()) {
+			return original.call(effectInstance);
+		}
+
+		if (!ServerMobEffectEvents.ALLOW_ADD.invoker().allowAdd(effectInstance, (LivingEntity) (Object) this, MobEffectUtil.getCommandContext())) {
+			return false;
+		} else {
+			return original.call(effectInstance);
+		}
+	}
+
+	@Inject(
+			method = "addEffect(Lnet/minecraft/world/effect/MobEffectInstance;Lnet/minecraft/world/entity/Entity;)Z",
+			at = @At(
+					value = "INVOKE",
+					target = "Ljava/util/Map;get(Ljava/lang/Object;)Ljava/lang/Object;"
+			)
+	)
+	private void beforeAddEffect(MobEffectInstance effectInstance, Entity entity, CallbackInfoReturnable<Boolean> cir) {
+		if (this.fabric_isClient()) {
+			return;
+		}
+
+		ServerMobEffectEvents.BEFORE_ADD.invoker().beforeAdd(effectInstance, (LivingEntity) (Object) this, MobEffectUtil.getCommandContext());
+	}
+
+	@Inject(
+			method = "forceAddEffect",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/world/entity/LivingEntity;canBeAffected(Lnet/minecraft/world/effect/MobEffectInstance;)Z",
+					shift = At.Shift.AFTER
+			)
+	)
+	private void beforeForceAddEffect(MobEffectInstance effectInstance, Entity entity, CallbackInfo ci) {
+		if (this.fabric_isClient()) {
+			return;
+		}
+
+		ServerMobEffectEvents.BEFORE_ADD.invoker().beforeAdd(effectInstance, (LivingEntity) (Object) this, MobEffectUtil.getCommandContext());
+	}
+
+	@Inject(
+			method = "onEffectAdded",
+			at = @At("RETURN")
+	)
+	private void afterAddEffect(MobEffectInstance effectInstance, Entity entity, CallbackInfo ci) {
+		if (this.fabric_isClient()) {
+			return;
+		}
+
+		if (!(this).level().isClientSide()) {
```

Comment:
is this not always true?

Reply:
It is 
<img width="33" height="56" alt="552977803633623040-1" src="https://github.com/user-attachments/assets/0e61e9cc-f29b-4087-bf08-8d552d6102e2" />